### PR TITLE
web: daterange not responsive

### PR DESCRIPTION
**Pull request by msh-odoo**
Task: https://www.odoo.com/web#id=2117229&action=333&active_id=1251&model=project.task&view_type=form&menu_id=4720

Pad: https://pad.odoo.com/p/r.1801f084f56387a11a27a1e02832543f

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/fields/basic_fields.js [hunk 1]
```diff
@@ -566,6 +566,10 @@ var LinkButton = AbstractField.extend({
 
 var FieldDateRange = InputField.extend({
     className: 'o_field_date_range',
+    events: {
+        'show.daterangepicker': '_onDateRangePickerShow',
+        'hide.daterangepicker': '_onDateRangePickerHide',
+    },
```
Comment (aab-odoo) [inline addons/web/static/src/js/fields/basic_fields.js]:
```
events: _.extends({}, InputField.prototype.events, {
   // your events
}),
```

File: addons/web/static/src/js/fields/basic_fields.js [hunk 2]
```diff
@@ -677,6 +684,36 @@ var FieldDateRange = InputField.extend({
...
+            window.removeEventListener('scroll', this._onScroll, true);
+        }
+    },
+    /**
+     * Reacts to the daterangepicker being shown
+     *
+     * @private
+     */
+    _onDateRangePickerShow() {
+        var self = this;
```
Comment (aab-odoo) [inline addons/web/static/src/js/fields/basic_fields.js]: you can get rid of self if you use an arrow function

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 3]
```diff
@@ -2998,6 +2998,35 @@ QUnit.module('basic_fields', {
...
+                    '<field name="datetime" widget="daterange" options="{\'related_end_date\': \'datetime_end\'}"/>' +
+                    '<field name="datetime_end" widget="daterange" options="{\'related_start_date\': \'datetime\'}"/>' +
+                '</form>',
+            res_id: 1,
+        });
+
+        await testUtils.form.clickEdit(form);
+        await testUtils.dom.click(form.$('.o_field_date_range:first'));
+
+        assert.isVisible($('.daterangepicker:first'), "date range picker should be closed");
```
Comment (aab-odoo) [inline addons/web/static/tests/fields/basic_fields_tests.js]: be opened

File: addons/web/static/src/js/libs/daterangepicker.js [hunk 4]
```diff
@@ -10,5 +10,18 @@ daterangepicker.prototype.clickDate = function (ev) {
         clickDateFunction.apply(this, arguments);
     }
 };
+/**
+ * Compute daterangepicker to open up or down based on top/bottom space in window.
+ */
+const moveFunction = daterangepicker.prototype.move;
+daterangepicker.prototype.move = function (ev) {
+    const offset = this.element.offset();
+    if (offset.top + this.container.height() * 1.5 >= $(window).height() + $(window).scrollTop() && this.container.height() + this.element.outerHeight() < offset.top) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/libs/daterangepicker.js]: Could you explain why `this.container.height() < offset.top` isn't enough?

Comment (msh-odoo) [reply addons/web/static/src/js/libs/daterangepicker.js]: I don't remember why I did like this :)

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by MiquelRForgeFlow** [comment]
this PR should be fixed, runbot is red

**Pull request closed without merging**